tensorflow/core/kernels/adjust_hue_op_gpu.cu.cc

```diff
@@ -0,0 +1,182 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#if GOOGLE_CUDA
+
+#define EIGEN_USE_GPU
+
+#include "adjust_hue_op.h"
+#include "tensorflow/core/util/cuda_kernel_helper.h"
+
+namespace tensorflow {
+
+typedef Eigen::GpuDevice GPUDevice;
+
+namespace internal {
+
+typedef struct RgbTuple {
+
+  float r, g, b;
+
+} RgbTuple;
+
+typedef struct HsvTuple {
+
+  float h, s, v;
+
+} HsvTuple;
+
+__device__ HsvTuple rgb2hsv_cuda(const float r, const float g, const float b) {
+
+  HsvTuple tuple;
+
+  const float M = fmaxf(r, fmaxf(g, b));
+  const float m = fminf(r, fminf(g, b));
+  const float chroma = M - m;
+
+  float h = 0.0f, s = 0.0f;
+
+    // hue
+    if (chroma > 0.0f) {
+      if (M == r) {
+
```

Comment:
Consistently remove empty lines at the beginning and end of each block. 

In general, TensorFlow wants to fit more content within one page. Adding new line only between important functional blocks. There is no need for cosmetic new lines.